LCM/scripts/GetDscConfiguration.py

```diff
@@ -2,22 +2,53 @@
 import fileinput
 import sys
 import subprocess
+import json
+import time
+import datetime
+import os
+import os.path
+from OmsConfigHostHelpers import write_omsconfig_host_telemetry, write_omsconfig_host_event
+
+from imp                import load_source
+from os.path            import dirname, isfile, join, realpath
+from fcntl              import flock, LOCK_EX, LOCK_UN
+
+pathToCurrentScript = realpath(__file__)
+pathToCommonScriptsFolder = dirname(pathToCurrentScript)
+
+helperLibPath = join(pathToCommonScriptsFolder, 'helperlib.py')
+helperlib = load_source('helperlib', helperLibPath)
 
 omi_bindir = "<CONFIG_BINDIR>"
 omicli_path = omi_bindir + "/omicli"
+dsc_host_base_path = '/opt/dsc'
+dsc_host_path = join(dsc_host_base_path, 'bin/dsc_host')
+dsc_host_output_path = join(dsc_host_base_path, 'output')
+dsc_host_lock_path = join(dsc_host_base_path, 'dsc_host_lock')
+dsc_host_switch_path = join(dsc_host_base_path, 'dsc_host_ready')
+
+if ("omsconfig" in helperlib.DSC_SCRIPT_PATH):
+    write_omsconfig_host_event(pathToCurrentScript, isfile(dsc_host_switch_path))
+
+if ("omsconfig" in helperlib.DSC_SCRIPT_PATH) and (isfile(dsc_host_switch_path)):
+    is_oms_config = True
+else:
+    is_oms_config = False
 
 parameters = []
-parameters.append(omicli_path)
-parameters.append("iv")
-parameters.append("<DSC_NAMESPACE>")
-parameters.append("{")
-parameters.append("MSFT_DSCLocalConfigurationManager")
-parameters.append("}")
-parameters.append("GetConfiguration")
-
-#s = ""
-#for param in parameters:
-#    s += param + " "
+
+if is_oms_config:
+    parameters.append("/opt/dsc/bin/dsc_host")
```

Comment:
>/opt/dsc/bin/dsc_host [](start = 23, length = 21)

we should use omsconfig path to store host binary? /opt/microsoft/omsconfig/bin

Reply:
omsconfig is a different package. so it is better to separate everything.

---
In reply to: [276379566](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/553#discussion_r276379566) [](ancestors = 276379566)